src/app/modules/group/components/user-list/user-list.component.ts

```diff
@@ -8,43 +10,48 @@ import { GetGroupMembersService, Member } from '../../http-services/get-group-me
   templateUrl: './user-list.component.html',
   styleUrls: [ './user-list.component.scss' ]
 })
-export class UserListComponent implements OnChanges, OnDestroy {
+export class UserListComponent implements OnChanges {
 
   @Input() group? : Group;
-  state: 'loading' | 'error' | 'empty' | 'ready' = 'loading';
+  state: 'loading' | 'error' | 'ready' = 'loading';
 
-  members: Member[] = [];
+  currentSort: string[] = [];
 
-  private subscription?: Subscription;
+  members: Member[] = [];
 
-  constructor(private getGroupMembersService: GetGroupMembersService) { }
+  private dataFetching = new Subject<string[]>();
+
+  constructor(private getGroupMembersService: GetGroupMembersService) {
+    this.dataFetching.pipe(
+      switchMap(sort => {
+        if (this.group) {
+          if (sort === []) this.state = 'loading';
+
+          return this.getGroupMembersService.getGroupMembers(this.group.id, sort);
+        } else {
+          throw new Error('group is null');
+        }
+      })
+    ).subscribe(
+      members => {
+        this.members = members;
+        this.state = 'ready';
+      },
+      _err => {
+        this.state = 'error';
+      });
+  }
 
   ngOnChanges(_changes: SimpleChanges): void {
-    this.reloadData();
+    this.dataFetching.next([]);
```

Comment:
You should use the `reset` method on the table, see https://www.primefaces.org/primeng/showcase/#/table

Reply:
fixed